components/Rules.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div>
+    <div ref="iframeContainer" class="iframe-container" />
+    <dl>
+      <dt>Results</dt>
+      <dd ref="results" />
+    </dl>
+  </div>
+</template>
+
+<script>
+import ruleEnginesPerEnv from '../services/ruleEngines.json';
+import { featurePolicy } from '../services/config';
+
+export default {
+  mounted () {
+    const origin = window.location.origin;
+    const rules = ruleEnginesPerEnv[process.env.NODE_ENV];
+    const ruleEngines = {};
+
+    const iframeReady = [];
+    const iframeReadyCb = [];
+    const addIframePromise = () => { iframeReady.push(new Promise((resolve) => { iframeReadyCb.push(resolve) })) };
+    addIframePromise();
+
+    window.addEventListener("message", (event) => {
+      // console.log(event);
+      if (event.origin !== origin) {
+        return;
+      }
+      const results = this.$refs.results;
+      switch (event.data.event) {
+      case 'ready':
+        iframeReadyCb.pop()();
+        break;
+      case 'isElegibleResult':
+        for (const sentEvent of event.data.events || []) {
+          results.innerText += `${sentEvent.type}: ${JSON.stringify(sentEvent.params, null, 2)}\n`;
+        }
+        break;
+      }
+
+    }, false);
+
+    const ready = () => {
+
+      const iframeContainer = this.$refs.iframeContainer;
+
+      for (const rule of rules) {
+        const iframe = document.createElement('iframe');
+        iframe.allow = featurePolicy.join('; ');
+        iframe.src = rule;
+        addIframePromise();
+        iframeContainer.appendChild(iframe);
+        ruleEngines[rule] = iframe;
+      }
+      iframeReadyCb.pop()();
+    }
+
+    document.addEventListener('DOMContentLoaded', ready, false);
+
+    Promise.all(iframeReady).then(() => {
+      for (const urlOrigin in ruleEngines) {
+        ruleEngines[urlOrigin].contentWindow.postMessage({
+          event: 'isElegible',
```

Comment:
can we fix the spelling of 'eligible' everywhere? It has two I's -> `isEligible`